File: server/src/main/java/com/kruna1pate1/pictionaryserver/dto/RoomDto.java
package com.kruna1pate1.pictionaryserver.dto;

import com.kruna1pate1.pictionaryserver.model.Player;
import com.kruna1pate1.pictionaryserver.model.Round;
import com.kruna1pate1.pictionaryserver.model.enums.GameStatus;
import com.kruna1pate1.pictionaryserver.model.enums.ServerCode;
import lombok.Data;

import java.util.Map;

/**
 * Created by KRUNAL on 20-05-2022
 */
@Data
public class RoomDto {

    private String roomId;

//    private ServerCode responseCode;

    private String name;

    private Integer capacity;

    private PlayerDto[] players;

    private GameStatus status;

//    private Round round;
    private Integer playerCount;

}

File: client/mobile/app/src/main/java/com/kruna1pate1/pictionaryapp/data/dto/RoomDto.kt
package com.kruna1pate1.pictionaryapp.data.dto

import com.kruna1pate1.pictionaryapp.domain.model.enums.GameStatus
import com.kruna1pate1.pictionaryapp.domain.model.Room

data class RoomDto(
    val roomId: String,
    val name: String,
    val playerCount: Int,
    val capacity: Int,
    val status: GameStatus
) {

    fun toRoom(): Room {

        return Room(
            roomId = roomId,
            name = name,
            playerCount = playerCount,
            capacity = capacity,
            status = status
        )
    }
}
